### PR TITLE
(feat) adding pre-defined custom template to slack plugin

### PR DESCRIPTION
**Pull request by eoinmcafee00**

### CODE REVIEW

**Review by jimsheldon — CHANGES_REQUESTED** [review]

**Review by jimsheldon — APPROVED** [review]

File: plugin.go [hunk 1]
```diff
@@ -167,8 +264,7 @@ func (p Plugin) Exec() error {
 		options := []slack.MsgOption{}
 		if text != "" {
 			options = append(options, slack.MsgOptionText(text, false))
-		}
-		if len(blocks) > 0 {
+		} else if len(blocks) > 0 {
```
Comment (tphoney) [inline plugin.go:267]: what happens if there are no blocks ?
what should it fall through to ?

File: main.go [hunk 2]
```diff
@@ -240,26 +250,28 @@ func run(c *cli.Context) error {
...
+			CustomTemplate: c.String("customTemplate"),
 		},
 	}
 	if plugin.Build.Commit == "" {
 		plugin.Build.Commit = "0000000000000000000000000000000000000000"
 	}
-	if plugin.Config.Webhook == "" {
-		return errors.New("Missing webhook")
+	if plugin.Config.Webhook == "" && plugin.Config.AccessToken == "" {
+		return errors.New("webhook or access token required")
```
Comment (tphoney) [inline main.go:274]: the logic does not match the text

**Review by tphoney — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (eoinmcafee00) [reply main.go]: done

File: plugin.go (hunk 1 above)
Comment (eoinmcafee00) [reply plugin.go]: fixed.

**Review by tphoney — APPROVED** [review]

**Pull request merged**